guides/documentations/essentials.md

```diff
@@ -0,0 +1,265 @@
+It uses ANTLR or Lidy grammars.
+The DefaultParser provides these methods:
+
+- `parse`: Convert the content of files into Components.
+- `getModels`: Get the list of model paths from all parsable files.
+- `parseConfiguration`: Set configuration into Components.
+- `isParsable`: Indicate if this parser can parse this file. By default, this function return false only on null/undefined fileInformation.
+
+## Drawer
+
+The Drawer is used to define/override the way components are drawn on the screen.
+By default, the DefaultDrawer use D3 to draw. Only the `draw` function is called by Leto-Modelizer and used to draw all Components and ComponentLinks. You can override this method if you want to use another library.
+
+### Customize your drawer models
+
+By default, we provide a default icon and model for the drawer.
+
+If you don't want to use them, override them with your svg. See [template.md](https://github.com/ditrit/leto-modelizer-plugin-core/blob/main/guides/svg/template.md).
+
+#### Use custom model in drawer
+
+You can provide multiple models and icons that can be used by the drawer.
+
+To do that, you have to add the svg of your icons in the folder `public/icons` and the svg of your models in the folder `public/models`.
+
+## DefaultData
+
+The DefaultData class exposes all the following data:
+- Plugin configuration
+- Plugin name
+- Plugin version
+- Components array
+- Variables array
+- All Component definitions
+- All Component link definitions
+- Parse errors array
+- Event manager that contains the function to emit event
+
+So this class also contains all the methods for managing these data (get, add, remove etc...):
+
+- `getComponentById`: Get component by id.
+- `renameComponentId`: Rename a component ID and update all its occurence inside link/References attribute type.
+- `getComponentsByType`: Get all components corresponding to the given type.
+- `addComponent`: Create and add new component inside components list.
+- `generateComponentId`: Generate id from definition and components list.
+- `removeComponentById`: Remove component by id and all attributes that used this component id.
+- `removeLink`: Remove link attribute in components.
+- `getLinks`: Get all links from all component attributes.
+- `getAttributeValue`: Get the value of an attribute.
+- `getLinkedComponentsIds`: Get the ID of the linked component.
+- `getVariableValue`: Get the value of a variable.
+- `setVariableValue`: Set the value of a variable.
+- `getWorkflowLinks`: Build internal links for workflow containers.
+- `getUsedLinkDefinitions`: Uniquely get the definitions used for existing links.
+- `initLinkDefinitions`: Initialize all link definitions from all component attribute definitions.
+- `getChildren`: Get children of container component with corresponding id.
+- `getEventLogById`: Get event log by id.
+- `emitEvent`: Emit event with log.
+
+## DefaultConfiguration
+
+The DefaultConfiguration class is used to set your Plugin's configuration.
```

Comment:
prefere to have a separate md file for all configuration explanation.

Reply:
done